cmd/sanitize-prow-jobs/main.go

```diff
@@ -6,36 +6,126 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 	"sync"
 
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	prowconfig "k8s.io/test-infra/prow/config"
 	"sigs.k8s.io/yaml"
 )
 
-const defaultCluster = "api.ci"
-
 type options struct {
 	prowJobConfigDir string
+	configPath       string
 
 	help bool
 }
 
+// Config is the structure of configuration file for this tool
+type Config struct {
+	Clusters Clusters
+}
+
+// Clusters is the cluster parameter for each Prow job, i.e., where it runs
+type Clusters struct {
+	// the cluster context name if no other condition matches
+	Default string
+	// the cluster context name for non Kubernetes jobs
+	NonKubernetes string
+
+	Groups []Group
+	Files  []File
+}
+
+// Group is where a group of jobs run
+type Group struct {
+	// name of the group
+	Name string
+	// the job names
+	Jobs []string
+	// the cluster where the jobs run
+	Cluster string
```

Comment:
Why does this field exist? The group should just have a name and a list of jobs. The `Cluster` then specifies the `group(s)`.

Same comment for the `File` struct below.

Reply:
So you want to define cluster on groups instead of each group?

Reply:
```yaml
default: "foo"
groups:
  "bar":
    paths:
    - etc
    - what
    jobs:
    - thing
    - other
```

This seems like the simplest approach. no?

Reply:
Will do this tonight.

Reply:
Asked my question
https://coreos.slack.com/archives/GB7NB0CUC/p1587421714096000

Reply:
Done.